phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -225,16 +232,45 @@ private static boolean isMultiPortUrl(String portStr) {
             }
             return false;
         }
-        
+
         public static ConnectionInfo create(String url) throws SQLException {
             url = url == null ? "" : url;
             if (url.isEmpty() || url.equalsIgnoreCase("jdbc:phoenix:")
                     || url.equalsIgnoreCase("jdbc:phoenix")) {
                 return defaultConnectionInfo(url);
             }
+
+            // ex: jdbc:phoenix+hrpc:hostname1....
             url = url.startsWith(PhoenixRuntime.JDBC_PROTOCOL)
                     ? url.substring(PhoenixRuntime.JDBC_PROTOCOL.length())
                     : PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR + url;
+
+            HBaseRegistryBootstrapType bootstrapAsEnum = null;
+
+            // ex: +hrpc:hostname1....
+            if (url.startsWith(String.valueOf(PhoenixRuntime.JDBC_PROTOCOL_CONNECTOR_PREFIX))) {
+                int offset = url.indexOf(PhoenixRuntime.JDBC_PROTOCOL_SEPARATOR);
+                final String firstToken = url.substring(0, offset);
+
+                // ex: +hrpc
+                final String bootstrapString = firstToken
+                        .replace(String.valueOf(PhoenixRuntime.JDBC_PROTOCOL_CONNECTOR_PREFIX), "")
+                        .toUpperCase();
+
+                if (Strings.isNullOrEmpty(bootstrapString)) {
+                    throw getMalFormedUrlException(url);
```

Comment:
IUC this will not allow URLs without bootstraps specified.
We should support bootstrap-less URLs, and default to ZK for backwards compatibility.

Reply:
@stoty are you saying a URL like `+:hostname1....` is a real possibility? I think that's a malformed string. If it was bootstrap-less it would be  `jdbc:phoenix:hostname1`

Reply:
Right, I didn't notice this block is only for URLS that have the "+" character.